odoo/addons/base/models/ir_fields.py

```diff
@@ -349,7 +349,7 @@ def db_id_for(self, model, field, subfield, value):
             field_type = _(u"name")
             if value == '':
                 return False, field_type, warnings
-            flush()
+            flush(field.comodel_name)
```

Comment:
Please add an explicit parameter name, because a valid XML id *can* be a valid model name, too.
Something like `flush(model=...)` wouldn't hurt.

Reply:
That's an option I didn't take because I didn't want to change the calling conventions for `flush` as it's a callback passed through the context with the possibility of more or less anyone messing with it. No problem changing it if we don't think that's a concern.

Reply:
> it's a callback passed through the context with the possibility of more or less anyone messing with it
Yes, but developers are not expected to mess with it.
So please add an explicit parameter for models.